src/apps/api/views/competitions.py

```diff
@@ -212,45 +216,42 @@ def email_all_participants(self, request, pk):
         batch_send_email.apply_async((comp.pk, content))
         return Response({}, status=status.HTTP_200_OK)
 
-    def collect_leaderboard_data(self, competition):
-        # TODO: Need to differentiate between leaderboards on different phases
-        #  (after there are different leaderboards on each phase)
-        # Maybe: Add the ability to sort submissions by score
-
-        # Query Needed data and filter to what is needed.
-        phase_pks = [phase.id for phase in Phase.objects.filter(competition_id=competition.id)]
-        submission_query = Submission.objects.filter(
-            Q(phase_id__in=phase_pks) & Q(has_children=False) & Q(leaderboard_id__isnull=False))
-        if not submission_query.exists():
-            raise ValidationError("There are no submissions on the leaderboard")
-
-        # Build the data needed for the leaderboard_data's into a dictionary
-        leaderboard_data = {}
-        leaderboard_titles = {}
-        columns_titles = {}
-        for submission in submission_query:
-            leaderID = submission.leaderboard_id
-            if leaderID not in leaderboard_titles.keys():
-                leaderboard = Leaderboard.objects.prefetch_related('columns').get(pk=leaderID)
-                columns_titles.update({leaderID: {}})
-                leaderboard_titles.update({leaderID: f"{leaderboard.title}({leaderID})"})
-                leaderboard_data[leaderboard_titles[leaderID]] = {}
-            columns_titles[leaderID].update({f'col:{col.id}-task:{submission.task_id}': f'{col.title}-{col.id}(Task:{submission.task_id})' for col in leaderboard.columns.all()})
-            if submission.owner.username not in leaderboard_data[leaderboard_titles[leaderID]].keys():
-                leaderboard_data[leaderboard_titles[leaderID]][submission.owner.username] = {}
-            for score in submission.scores.all():
-                leaderboard_data[leaderboard_titles[leaderID]][submission.owner.username].update({columns_titles[leaderID][f'col:{score.column_id}-task:{submission.task_id}']: float(score.score)})
+    def collect_leaderboard_data(self, competition, phase_pk=None):
+        if phase_pk:
+            phase = get_object_or_404(competition.phases.all(), id=phase_pk)
+            submission_query = [self.get_serializer(phase).data]
+            phase_id = phase.id
+        else:
+            phases = competition.phases.all()
+            submission_query = self.get_serializer(phases, many=True).data
+            phase_id = phases.first().id
```

Comment:
I think this will do an additional query, may want to do..
```suggestion
            phase_id = phases[0].id
```

but before we access `[0]` index may want to assert/raise validation error if len == 0